cockatrice/src/chatview.cpp

```diff
@@ -151,239 +156,236 @@ void ChatView::appendMessage(QString message, QString sender, UserLevelFlags use
-                
-                appendCardTag(cursor, cardName);
-            }
-            else if (message.startsWith("[url]"))
-            {
-                message = message.mid(5);
-                int closeTagIndex = message.indexOf("[/url]");
-                QString url = message.left(closeTagIndex);
-                if (closeTagIndex == -1)
-                    message.clear();
-                else
-                    message = message.mid(closeTagIndex + 6);
-                
-                appendUrlTag(cursor, url);
-            }
-            else
-            {
-                // Not a valid tag
-                cursor.insertText("[", defaultFormat);
-                message = message.mid(1);
-            }
+            appendUrlTag(cursor, fullWordUpToSpaceOrEnd);
+            cursor.insertText(rest, defaultFormat);
+            return;
         }
-        else if (index == urlFirstIndex) // The message now starts with either: www. , http:// , or https://
+    }
+
+    // check word mentions
+    foreach (QString word, highlightedWords)
+    {
+        if (fullWordUpToSpaceOrEnd.compare(word, Qt::CaseInsensitive) == 0)
         {
-            int urlEndIndex = message.indexOf(phraseEnder, 0);
-            if (urlEndIndex == -1)
-                urlEndIndex = message.size();
-            QString urlText = message.left(urlEndIndex);
-            QUrl qUrl(urlText);
-            if (qUrl.isValid())
-                appendUrlTag(cursor, urlText);
-            else
-                cursor.insertText(urlText);
-            if (urlEndIndex == -1)
-                message.clear();
-            else
-                message = message.mid(urlEndIndex);
+            // You have received a valid mention of custom word!!
+            highlightFormat.setBackground(QBrush(getCustomHighlightColor()));
+            highlightFormat.setForeground(settingsCache->getChatHighlightForeground() ? QBrush(Qt::white) : QBrush(Qt::black));
+            cursor.insertText(fullWordUpToSpaceOrEnd, highlightFormat);
+            cursor.setCharFormat(defaultFormat);
+            cursor.insertText(rest, defaultFormat);
+            QApplication::alert(this);
+            return;
         }
-        else if (index == mentionFirstIndex)
-        {
-            int firstSpace = message.indexOf(" ");
-            QString fullMentionUpToSpaceOrEnd = (firstSpace == -1) ? message.mid(1) : message.mid(1, firstSpace - 1);
-            QString mentionIntact = fullMentionUpToSpaceOrEnd;
+    }
 
-            if ((!mentionEnabled && !isModeratorSendingGlobal(userLevel, fullMentionUpToSpaceOrEnd)) || tabSupervisor->getIsLocalGame())
-            {
-                cursor.insertText("@");
-                message = message.mid(1);
-            }
-            else
-            {
-                QMap<QString, UserListTWI *> userList = tabSupervisor->getUserListsTab()->getAllUsersList()->getUsers();
+    // not a special word; just print it
```

Comment:
We could check for special strings within by potentially nesting the function within itself, allowing us to extract things such as `(([[card name]]))` as `card name` instead of just text.